octoprint_filamentsensorsimplified/__init__.py

```diff
@@ -109,6 +120,17 @@ def get_disable(self):
 
 	# test pin value, power pin or if its used by someone else
 	def on_api_command(self, command, data):
+		if command == "pollStatus":
```

Comment:
I've got an idea for how to simplify this but it will mean more code change which to be honest I'm quite afraid of because of potential bug creation - don't react to rising/falling edge after the print starts, but react to it after the plugin setup - this way we can get rid of polling. For the client icon update we could use _self._plugin_manager.send_plugin_message_ with enhanced data of boolean type, which is basically HTTP PUSH which is exactly what we need, say it can be called _updateIconOnly_, this would specify if the notification should popup (in case of running print) or we want to only update the state of icon. This would also make the change visible instantaneously on the UI. If you feel like it youcan rework it if don't there's high probability I will rework it later.

Reply:
Did a bit of testing on it. Its possible but I didn't want to rewrite it all and the load of constantly  pulling the sensor might be heavy.

But at some point I will probably rewrite it like that and use the new libraries

Reply:
is it possible to change the pull request to merge into develop branch? I would accept the pull request and simplify it as I wrote